### ISSUE TITLE
RqMultipart class unreadable

### ISSUE THREAD

**Issue by CarlosGines** [yegor256/takes#651]
Class `RqMultipart`, apart from being quite long, is confusing and extremely hard to read and understand. It makes it really difficult to maintain. 

Please refactor it so that it becomes more understandable. Split it into more classes or whatever it is needed for that purpose.

**Comment by davvd** [comment]
@yegor256 please do something about it, see [par.21](http://at.teamed.io/policy.html#21)

**Comment by davvd** [comment]
@carlosgines since there is no milestone yet I set it to "1.0"

**Comment by davvd** [comment]
@triberraar this task is yours, please help (see our key [principles of work](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html)). If you have any technical questions, don't hesitate to ask right here... The budget of this issue is **30 mins**, which is exactly how much will be paid when the task is *done* (see [this](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) for explanation)

**Comment by davvd** [comment]
@carlosgines thank you for the ticket reported, I topped your account for 15 mins, transaction `AP-2S985733VJ2962257`

**Comment by davvd** [comment]
@triberraar check this ["no obligations principle"](http://www.xdsd.org/2014/04/13/no-obligations-principle.html).. This task is on your name for at least 15 days. If you can't close it within the next 48 hours I will have to assign someone else to it. [This article](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html) should help if you're stuck

-30 to your rating, your total score is [-90](http://www.netbout.com/b/36486?open=rating)

**Comment by davvd** [comment]
@triberraar it takes too long. Usually, we expect any task to be finished in less than a week. I'll assign someone else. Please stop working with it right now. See our [no obligations](http://www.xdsd.org/2014/04/13/no-obligations-principle.html) principle... -60 to your rating, your total score is [-270](http://www.netbout.com/b/36486?open=rating)

**Comment by davvd** [comment]
@essobedo the task is yours please proceed

**Comment by essobedo** [comment]
@davvd this issue depends on #626

**Comment by davvd** [comment]
> @davvd waiting for #626 

@essobedo agreed, we'll wait for #626

**Comment by essobedo** [comment]
@davvd here is my PR #665 please find a reviewer.

**Comment by davvd** [comment]
> @davvd here is my PR #665 please find a reviewer.

@essobedo OK

**Comment by davvd** [comment]
@carlosgines #626 is closed, we were waiting for it.. maybe we can continue?

**Comment by essobedo** [comment]
@davvd I confirm that since #626 is closed, I can continue, I could even propose a PR as you can see above

**Comment by essobedo** [comment]
@CarlosGines the PR has been merged, you can close the ticket, thx in advance

**Comment by CarlosGines** [comment]
@essobedo closed, thank you.

**Comment by essobedo** [comment]
@davvd The ticket is ready for the QA, thx in advance

**Comment by essobedo** [comment]
@davvd a QA validation is needed here, thx

**Comment by davvd** [comment]
> @davvd I confirm that since #626 is closed, I can continue, I could even propose a PR as you can see above 

@essobedo OK

**Comment by davvd** [comment]
@elenavolokhova please, let us know what do you think about this ticket, according to our [QA rules](http://at.teamed.io/qa.html)

**Comment by davvd** [comment]
@carlosgines there is a puzzle in this code `651-43df2f64`/#666, we'll resolve it later

**Comment by elenavolokhova** [comment]
@davvd Looks good!

**Comment by davvd** [comment]
> @davvd Looks good!

@elenavolokhova many thanks

**Comment by davvd** [comment]
@essobedo 10 mins was added to the account of @elenavolokhova (for QA review), in transaction 84361996... thanks, I just added **30 mins** to your account, payment `84362005`, 89 hours and 12 mins spent... +30 added to your rating, current score is: [+525](http://www.netbout.com/b/36421?open=rating)

**Comment by davvd** [comment]
@carlosgines the last puzzle `651-43df2f64`/#666 originated from here solved
